src/xb-machine.c

```diff
@@ -1406,7 +1406,13 @@ xb_machine_func_eq_cb (XbMachine *self,
 	}
 
 	/* should have been checked above */
-	g_assert_not_reached ();
+	g_set_error (error,
```

Comment:
This shouldn’t be necessary. All the appropriate checking should have been done by `xb_machine_check_two_args()` at the start of the function. Something must be wrong there.

Reply:
I never got to this point, I just changed it to match the others. I agree it probably can't be reached, but I didn't like the other `g_assert_not_reached()` just exploding when running under afl.

Reply:
Which one was the `g_assert_not_reached()` which was exploding under afl? That shouldn’t be happening — is an `xb_machine_check_two_args()` (or similar) missing in that case?

Reply:
I believe it was a `ne` with `TEXT:INTE` opcodes.